agent/versioner/versioner.go

```diff
@@ -135,3 +135,13 @@ func (v *Versioner) XbcloudVersion() (string, error) {
 func (v *Versioner) Qpress() (string, error) {
 	return v.binaryVersion(qpressBin, 255, qpressRegexp)
 }
+
+// BinaryVersion retrieves agent binary version.
+func (v *Versioner) BinaryVersion(
+	binaryName string,
+	expectedExitCode int,
+	versionRegexp *regexp.Regexp,
+	arg ...string,
+) (string, error) {
+	return v.binaryVersion(binaryName, expectedExitCode, versionRegexp, arg...)
+}
```

Comment:
Why not just make `v.binaryVersion` public? This wrapper doesn't add any value, no?

Reply:
Now all Versioner's public methods depend on private `v.binaryVersion` and private `v.binaryVersion` easily can be extended

If private `binaryVersion` become public `BinaryVersion` then extend will be harder because you should extract to private again before extend

Reply:
That sounds pretty much like premature optimization.